lib/jwt.service.ts

```diff
@@ -14,9 +16,9 @@ export class JwtService {
     @Inject(JWT_MODULE_OPTIONS) private readonly options: JwtModuleOptions
   ) {}
 
-  sign(payload: string | Buffer | object, options?: jwt.SignOptions): string {
+  sign(payload: string | Buffer | object, options?: JwtSignOptions): string {
     const signOptions = this.mergeJwtOptions(
-      options,
+      { ...options },
```

Comment:
I spread it because I needed a copy of the object so I could remove the secret property in the mergeJwtOptions function without it affecting the original object.